src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -1,30 +1,46 @@
 from __future__ import annotations
 
 import abc
+import ast
+import typing
 from collections.abc import Callable, Iterable
+from enum import Enum
 
 from databricks.sdk.service.workspace import ObjectType, ObjectInfo, ExportFormat, Language
 from databricks.sdk import WorkspaceClient
 
 from databricks.labs.ucx.source_code.base import Advice, Deprecation
+from databricks.labs.ucx.source_code.python_linter import ASTLinter, PythonLinter
 from databricks.labs.ucx.source_code.whitelist import Whitelist, UCCompatibility
 
+if typing.TYPE_CHECKING:
+    from databricks.labs.ucx.source_code.site_packages import SitePackages, SitePackage
 
-class Dependency:
 
-    @staticmethod
-    def from_object_info(object_info: ObjectInfo):
-        assert object_info.object_type is not None
-        assert object_info.path is not None
-        return Dependency(object_info.object_type, object_info.path)
+class DependencyType(Enum):
+    WORKSPACE_NOTEBOOK = "WORKSPACE_NOTEBOOK"
```

Comment:
I suspect that site-package files would not be loaded via the Workspace, would they ? In which case, relying on a single loader per usage scenario would fail.
It's possible that we'll be able to simplify a bit in the end state, but it's very useful at this point to know what object we're dealing with.
We have specialized loaders already for the scenarios currently covered: WorkspaceLoader and PackageLoader

Reply:
there has to be no difference between:

1. Local File
2. Workspace File (it can be read as a local file)
3. Package File (it can be read as a local file)

The only difference between workspace notebook and local notebook is the loader.

Reply:
I've created https://github.com/databrickslabs/ucx/issues/1455 to address this.